src/player.cpp

```diff
@@ -4160,14 +4160,14 @@ void player::pause()
     search_surroundings();
 }
 
-void player::shout( std::string msg )
+void player::shout( std::string msg, const std::string &verbal_shout_id )
 {
-    int base = 10;
+    int base = 85;
```

Comment:
> I can add a cooldown effect to prevent this, like there already is for other npc effects. Would that address this concern?

Could work, but you'd have to test it if it works properly. Would be good if the NPC stopped running onto the spot after being talked to. And initiated talk mode by themselves instead of just waiting for the player to do that.

>  This PR can be used in the problematic situations you described

The problem is, it solves those problems only if you know what to search for and where. In the context of the LoS limit PR:
* Quest givers can still run off, you just have better ability to catch them if you know they are in range (which you don't know if LoS limits it)
* Does nothing to help with NPCs charging into zeds
* You'd have to "scan" for NPCs by shouting periodically to find hidden NPCs if they weren't shown on the map. And this would work better if you could do it without getting an entire horde every time you do that

For it to actually help with those issues, the NPCs would have to shout back.